compiler/rustc_target/src/asm/avr.rs

```diff
@@ -89,7 +114,24 @@ impl AvrInlineAsmReg {
         _arch: InlineAsmArch,
         _modifier: Option<char>,
     ) -> fmt::Result {
-        out.write_str(self.name())
+        use AvrInlineAsmReg::*;
+        let name = match self {
+            r25r24 => "r24",
+            r23r22 => "r22",
+            r21r20 => "r20",
+            r19r18 => "r18",
+            r17r16 => "r16",
+            r15r14 => "r14",
+            r13r12 => "r12",
+            r11r10 => "r10",
+            r9r8 => "r8",
+            r7r6 => "r6",
+            r5r4 => "r4",
+            r3r2 => "r2",
+            r1r0 => "r0",
```

Comment:
This function should emit the *exact* same string that LLVM's inline assembler will for the given register. It's not currently used by the backend, but will eventually be used for the fallback assembler where we generate a .S file and pass it to an external assembly (e.g. for the cranelift backend which doesn't support assembly natively).

Reply:
Understood.  I believe that would come from the `AsmName` field of the `llvm/Target/Register` class, so in this case it should be `r1:r0` (with the colon).  Will update.